### PR TITLE
Set Impact for collector flaws based on CVSS severity

### PR DESCRIPTION
**Pull request by jobselko**
This PR reworks and extends the CVSS section for collector flaws to be able to calculate the resulting impact. The impact is created based on the highest CVSS score. The PR also updates the OSV collector to collect CVSSv4.

Closes OSIDB-3487

### CODE REVIEW

File: collectors/utils.py [hunk 1]
```diff
@@ -98,3 +99,27 @@ def handle_urls(references: list, source_ref: str) -> list:
             pass
 
     return urls
+
+
+CVSS_SCORE_TO_IMPACT = {
```
Comment (jsvob) [inline collectors/utils.py:104]: Is this modeled after https://www.first.org/cvss/specification-document#Qualitative-Severity-Rating-Scale ?

**Review by jsvob — APPROVED** [review]
LGTM but I'm not knowledgeable in this area

File: collectors/utils.py (hunk 1 above)
Comment (jobselko) [reply collectors/utils.py:104]: Partially yes. It matches both Impact in OSIDB and CVSS severity defined by FIRST. However, FIRST does not define severity for CVSSv2, so I hope this simplification is good.

**Review by osoukup — APPROVED** [review]
LGTM

**Pull request merged**